cognify/optimizer/core/driver.py

```diff
@@ -107,17 +101,16 @@ def run(
         other_python_paths: Optional[list[str]] = None,
     ) -> tuple[float, list[tuple[TrialLog, str]], dict[str, TrialLog]]:
         self.build_tiered_optimization(evaluator)
-        first_layer_opt_config = self.layer_configs[0].opt_config
         logger.info("----------------- Start Optimization -----------------")
-        opt_cost, frontier, all_opt_logs = self.opt_layers[0].easy_optimize(
-            opt_config=first_layer_opt_config,
+        top_layer = self.opt_layer_factories[0]()
+        pareto_frontier = top_layer.exposed_optimize(
```

Comment:
I still don't get what `exposed_optimize` means?

Reply:
```
    def exposed_optimize(
        self,
        script_path: str,
        script_args: Optional[list[str]] = None,
        other_python_paths: Optional[list[str]] = None,
    ) -> list[tuple[TrialLog, str]]:
        """Trigger the entire optimization by calling this at the out-most layer
        """
```

This is mostly a wrapper of the `optimize` method.  Since the driver does not need `TopDownInformation` so I packaged it to provide a cleaner API.

Reply:
So this is like the entry point of optimization? Why not just call it optimizer_entry or optimizer_root?

Reply:
changed to `optimization_entry `